Architecture.md

```diff
@@ -28,7 +28,29 @@
 
 #### Client applications deployment
 
-***TODO: [INFRA] fill for the architecture used to deploy clients' applications***
+Pomegranate, the application and deployment manager, is in charge of monitoring and managing the containers' lifecycle, as well as configuring them.
+
+The [Rust](https://www.rust-lang.org/) programming language is used to develop the service. We used it because
+we wanted an efficient language that could be trusted with this job thanks to its compiled nature,
+and with support for the other technologies we use through the crates ecosystem.
```

Comment:
What do you think about this for talking about compiler constraints?
```suggestion
The [Rust](https://www.rust-lang.org/) programming language is used to develop the service. We used it because
we wanted an efficient and reliable language that could be trusted with this job thanks to its compiled nature and the restrictions it brings to drive us to write better code,
and with support for the other technologies we use through the crates ecosystem.
```